src/cpp/src/continuous_batching_pipeline.cpp

```diff
@@ -119,12 +151,13 @@ class ContinuousBatchingPipeline::Impl {
             can_use_partial_preemption = false;
         }
 
-        m_scheduler = std::make_shared<Scheduler>(updated_config, can_use_partial_preemption);
+        m_scheduler = std::make_shared<Scheduler>(updated_config, device_config.get_num_layers(), can_use_partial_preemption);
         // and finally create model runner
-        m_model_runner = std::make_shared<ModelRunner>(infer_request, updated_config);
+        m_model_runner = std::make_shared<ModelRunner>(infer_request, updated_config, device_config.get_num_layers(), true);
```

Comment:
should it be selected num_layer or 1 depending on is_need_per_layer_cache_control  ? similarly for Scheduler

Reply:
I put the `true` under the `is_use_cache_eviction` condition, but left the num_layers passed as-is, some future code may want to utilize the true number of layers in scheduler or runner irrespective of cache eviction settings.